js/views/modals/listingDetail/Listing.js

```diff
@@ -307,6 +308,22 @@ export default class extends BaseModal {
     });
   }
 
+  startPurchase() {
+    const selectedVariants = [];
+    this.variantSelects.each((i, select) => {
+      const variant = {};
+      variant.name = $(select).attr('name');
+      variant.value = $(select).val();
+      selectedVariants.push(variant);
+    });
+
+    this.purchaseModal = launchPurchaseModal({
+      listing: this.model.toJSON(),
```

Comment:
I only needed the data for the template. But you're right, there could be a case where the other functions are useful, and it can be converted to JSON in the render.